### PR TITLE
Update initialize domain playbook with register username code

### PR DESCRIPTION
**Pull request by theKosmoss**

### CODE REVIEW

File: scripts/register.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        domainName: '0x52616e6b6966792e697400000000000000000000000000000000000000000000', 
+        id: "0x65396635343931652d663037652d343039332d623665382d6339313238393700", 
+        name: '0x7468654b6f736d6f737300000000000000000000000000000000000000000000',
+        nonce: '0x00'
+    };
+
+    const registrarSignature = "0x933ebb2d720914dd53fccf4278d12a40e156e57f4a56873c3d5d7ccbe09c9015092b81179f82b3c2daf8192732bc844c22583aafe4634c47910979e0233b2e461c";
+
+    try {
+        const tx = await contract.register(
```
Comment (theKosmoss) [inline scripts/register.ts:39]: This works

File: .vscode/launch.json
Comment (peersky) [inline .vscode/launch.json:1]: Please don't commit .vscode files. We should keep our work IDE agnostic.

File: playbook/utils/deploy-to-local-anvil.sh [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #! /bin/bash
 
 rm -rf ./deployments/localhost
+rm -rf ./artifacts
```
Comment (peersky) [inline playbook/utils/deploy-to-local-anvil.sh:4]: I would also suggest removing `./cache`

File: scripts/register.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            registrarSignature,
+            {
+                name: ethers.utils.formatBytes32String(""),
+                id: ethers.utils.formatBytes32String(""),
+                domainName: ethers.utils.formatBytes32String(""),
+                wallet: ethers.constants.AddressZero,
+                targetDomain: ethers.utils.formatBytes32String(""),
+            }, 
+            ethers.constants.HashZero,
+            { maxFeePerGas: 100000000000, maxPriorityFeePerGas: 1000000000, value: 0 }
```
Comment (peersky) [inline scripts/register.ts:57]: Can we remove `maxFeePerGas` and `maxPriorityFeePerGas` ? I think they got here as debug values, we should not need them really.

**Review by peersky — CHANGES_REQUESTED** [review]
some changes desired

**Comment by peersky** [comment]
also needs changeset

File: playbook/initializeDomain.ts [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import { task, types } from 'hardhat/config';
 import { Multipass } from '../types';
-import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { LibMultipass } from '../types/src/Multipass';
+import crypto from "crypto";
+import { signRegistrarMessage } from "../playbook/utils/utils";
+import { ethers } from 'ethers';
```
Comment (peersky) [inline playbook/initializeDomain.ts:6]: I suggest using hre.ethers instead

File: playbook/initializeDomain.ts [hunk 5]
```diff
@@ -55,7 +65,42 @@ task('initializeDomain', 'Initialize domain name and activate it')
...
+      if (username) {
+        userAddress = userAddress ?? defaultPlayer;
+        const playerId = crypto.randomUUID().slice(0, 31);
+
+        const registrarMessage: LibMultipass.RecordStruct = {
+          wallet: userAddress,
+          name: hre.ethers.utils.formatBytes32String(username),
+          id: hre.ethers.utils.formatBytes32String(playerId),
+          domainName: hre.ethers.utils.formatBytes32String(domain),
+          validUntil: ethers.BigNumber.from(Math.floor(Date.now() / 1000) + (365 * 24 * 60 * 60)),
```
Comment (peersky) [inline playbook/initializeDomain.ts:78]: use hre.ethers instead

Comment (peersky) [inline playbook/initializeDomain.ts:78]:
```suggestion
          validUntil: hre.ethers.BigNumber.from(Math.floor(Date.now() / 1000) + (365 * 24 * 60 * 60)),
```

File: playbook/initializeDomain.ts (hunk 4 above)
Comment (peersky) [inline playbook/initializeDomain.ts:6]:
```suggestion
```

**Review by peersky — APPROVED** [review]
looks good, just optimize imports a bit

**Pull request merged**